app/mailers/application_mailer.rb

```diff
@@ -42,4 +42,10 @@ def group_mail(users, options)
   def set_locale_for(user)
     FastGettext.set_locale(user.locale.blank? ? "en" : user.locale)
   end
+
+  def set_url
+    unless (@url = URI.parse(Setting[:foreman_url])).present?
+      raise ":foreman_url is not set, please configure in the Foreman Web UI (More -> Settings -> General)"
```

Comment:
It seems that :foreman_url does accept a blank value.
![image](https://cloud.githubusercontent.com/assets/433583/6460081/e75ffdc6-c19c-11e4-8fcc-4ca3643cd0db.png)
So we're raising an error that it should be set.
Maybe it is a bug that you can set a blank value on :foreman_url